internal/version/version.go

```diff
@@ -1,8 +1,41 @@
 package version
 
+import (
+	"fmt"
+	"runtime/debug"
+)
+
 // GitCommit indicates which commit the rukpak binaries were built from
-var GitCommit string
+var (
+	gitCommit  string = "unknown"
+	commitDate string = "unknown"
+	repoState  string = "unknown"
+)
 
 func String() string {
-	return GitCommit
+	return fmt.Sprintf("revision: %q, date: %q, state: %q", gitCommit, commitDate, repoState)
+}
+
+func init() {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return
+	}
+	for _, setting := range info.Settings {
+		switch setting.Key {
+		case "vcs.revision":
+			gitCommit = setting.Value
+		case "vcs.time":
+			commitDate = setting.Value
+		case "vcs.modified":
+			switch setting.Value {
+			case "false":
+				repoState = "clean"
+			case "true":
+				repoState = "dirty"
+			default:
+				repoState = "unknown"
+			}
```

Comment:
super nit: is the switch here necessary? The go docs for the `vcs.modified` key mention it will be true or false - I'm interpreting that as the values will only ever be "true" or "false". Something a bit more simple:
```suggestion
if setting.Value != "true" {
    repoState = "clean"
} else {
    repoState = "dirty"
}
```

If we want to still use the switch, I would vote for just leaving the default case empty since the default value is already `"unknown"`.

That being said this is all just a nit about me not liking the poor readability of deep nesting and shouldn't block the PR from merging :smile:

Reply:
Looks nicer for sure :)